storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManagerTest.java

```diff
@@ -49,9 +45,10 @@ public class RemoteLogMetadataManagerTest {
 
     private final Time time = new MockTime(1);
 
-    @ParameterizedTest(name = "remoteLogMetadataManager = {0}")
-    @MethodSource("remoteLogMetadataManagers")
-    public void testFetchSegments(RemoteLogMetadataManager remoteLogMetadataManager) throws Exception {
+    private RemoteLogMetadataManager remoteLogMetadataManager = new TopicBasedRemoteLogMetadataManagerWrapperWithHarness();
+
+    @Test
+    public void testFetchSegments() throws Exception {
         try {
```

Comment:
It seems to me `TopicBasedRemoteLogMetadataManagerWrapperWithHarness` can be removed also. We don't use the wrapper actually. This test can be modified by following style:

```java
    @Test
    public void testFetchSegments() throws Exception {
        try (TopicBasedRemoteLogMetadataManagerHarness remoteLogMetadataManagerHarness = new TopicBasedRemoteLogMetadataManagerHarness()) {
            RemoteLogMetadataManager remoteLogMetadataManager = remoteLogMetadataManagerHarness.remoteLogMetadataManager();
```

noted `TopicBasedRemoteLogMetadataManagerHarness` needs to implement `AutoClosable`, and `remoteLogMetadataManager` should be a public method.